src/sentry.ts

```diff
@@ -143,19 +143,22 @@ export default class SentryService {
 
   async endSentryTransaction(write_command_out: boolean, error?: any): Promise<void> {
     if (this.app.config.environment === ENVIRONMENT.TEST) return;
+    try {
+      await this.updateSentryTransaction(error);
 
-    await this.updateSentryTransaction(error);
-
-    if (this.file_out && write_command_out) {
-      await this.writeCommandHistoryToFileSystem();
-    }
+      if (this.file_out && write_command_out) {
+        await this.writeCommandHistoryToFileSystem();
+      }
 
-    if (this.sentry_out) {
-      Sentry.getCurrentHub().getScope()?.getSpan()?.finish();
-      Sentry.getCurrentHub().getScope()?.getTransaction()?.finish();
-      if (error) {
-        Sentry.withScope(scope => Sentry.captureException(error));
+      if (this.sentry_out) {
+        Sentry.getCurrentHub().getScope()?.getSpan()?.finish();
+        Sentry.getCurrentHub().getScope()?.getTransaction()?.finish();
+        if (error) {
+          Sentry.withScope(scope => Sentry.captureException(error));
+        }
       }
+    } catch {
+      // If endSentryTransaction fails, don't error.
```

Comment:
Add some console.debug to these - an empty catch can be really confusing

Reply:
Added try/catch console.debug messages to all methods in SentryService, and one in BaseCommand sentry fails to initialize for currently unknown use cases.

```typescript
try {
  this.sentry = await SentryService.create(this.app, this.constructor as any);
} catch (e) {
  console.debug("SENTRY: an error occurred creating a new instance of SentryService");
}
if (!this.sentry) {
  console.debug("SENTRY: SentryService failed to initialize");
}
```